src/metadata.rs

```diff
@@ -111,18 +111,25 @@ impl Metadata {
     /// if let Ok(m) = "Life.on.Mars.(UK).S01E1E02.avi".parse::<Metadata>() {
     ///    assert_eq!(m.is_multi_episode(), true);
     ///    assert_eq!(m.episode(), Some(1));
+    ///    let episodes = m.episodes();
+    ///    assert_eq!(*episodes.last().unwrap(), 2);
     /// }
     ///
     /// if let Ok(m) = "Life.on.Mars.(UK).S01E1E02E03.avi".parse::<Metadata>() {
     ///    assert_eq!(m.is_multi_episode(), true);
     ///    assert_eq!(m.episode(), Some(1));
+    ///    let mut current_episode = m.episode().unwrap();
+    ///    for episode in m.episodes().iter() {
+    ///      assert_eq!(*episode, current_episode);
+    ///      current_episode = current_episode + 1;
+    ///    }
     /// }
     ///```
     pub fn is_multi_episode(&self) -> bool {
-        self.multi_episode.is_some()
+        !self.episodes.is_empty()
     }
-    pub fn episodes(&self) -> Option<&Vec<i32>> {
-        self.multi_episode.as_ref()
+    pub fn episodes(&self) -> Vec<i32> {
+        self.episodes.clone()
```

Comment:
I think this should return &Vec<i32> as this value will not change. Developers will just be borrowing the `Vec`

Reply:
if Episodes returns a `&Vec<i32`

We can implement a  convenience function as so
```rust
    pub fn last_episode(&self) -> Option<&i32> {
        if self.is_multi_episode() {
            self.episodes.last()
        } else {
            None
        }
    }
```

In theory we could deprecate `episode`or implement in the same manner. Though I would suggest changing its name to `first_episode`

Just an idea.

Reply:
I'd rather keep the interface least surprising, keep `episode` for most cases and provide `episodes` for the edge case so people can build whatever logic they want on top.

By the way, your snippet's `match` is useless. `self.episodes.last()` is equivalent to this. Or indeed `self.episodes().last()`.

Reply:
Ah you are correct. last() returns an Option. So I can fix that.
Of course keep episode so as not to break existing code usage.

Reply:
Agreed my code was redundant. I have addressed that in the final commit.
I have also ensured that there have been no user facing changes to the original code base. Everything works as previously.